h5netcdf/core.py

```diff
@@ -219,7 +225,10 @@ def __init__(self, parent, name):
         self._groups = _LazyObjectLookup(self, self._group_cls)
 
         for k, v in self._h5group.items():
-            if isinstance(v, h5py.Group):
+            if ((not no_h5pyd and isinstance(v, h5pyd.Group)) or
+                    isinstance(v, h5py.Group)):
```

Comment:
I think a cleaner way to write this would make a top level constant:
```
if no_h5pyd:
    h5_group_types = (h5py.Group,)
else:
    h5_group_types = (h5py.Group, h5pyd.Group)
```

Then use simply `isinstance(v, h5_group_types)`